### DISCUSSION TITLE
Is it possible to create my own SessionManagerLayer?

### DISCUSSION THREAD

**Discussion by xrtxn** [maxcountryman/tower-sessions#202] [Q&A]
I would like to store bearer tokens instead of cookies. As I understand it I can implement a way to use middlewares instead but this looks like a lot better way. My auth method is:
1. Basic login with username and password
2. Bearer response which the client will use for future requests

**Comment by maxcountryman** [comment]
Please see this discussion: https://github.com/maxcountryman/axum-login/discussions/214

**Reply by xrtxn** [reply]
I already saw the discussion, and understand it, I just said it would be easier to make a generic SessionManagerLayer so I can implement my own data types.

**Reply by maxcountryman** [reply]
You can use it directly if you like. That said, it's probably easier to use axum-login for this kind of thing.
